contents/tutorials/angular-surveys.md

```diff
@@ -0,0 +1,526 @@
+
+```JSON
+[
+   {
+     "id": "018cfcd5-107e-0000-49a1-8e7c6b825947",
+     "name": "Net promoter score (NPS) API Survey",
+     "description": "",
+     "type": "api",
+     "linked_flag_key": null,
+     "targeting_flag_key": null,
+     "questions": [
+       {
+         "type": "rating",
+         "scale": 10,
+         "display": "number",
+         "question": "How likely are you to recommend us to a friend?",
+         "description": "",
+         "lowerBoundLabel": "Unlikely",
+         "upperBoundLabel": "Very likely"
+       }
+     ],
+     "conditions": null,
+     "start_date": "2024-01-12T08:41:20.614000Z",
+     "end_date": null
+   }
+]
+```
+
+To fetch this object and integrate it with your survey UI, update your code in `app.component.ts`:
+
+```typescript file=app.component.ts
+import { Component, ChangeDetectorRef } from '@angular/core';
+import posthog from 'posthog-js'
+// rest of your imports
+
+@Component({
+  // your existing config
+})
+export class AppComponent implements OnInit {
+  title = 'angular-surveys';
+  showSurvey = true;
+  surveyTitle = 'Survey title';
+  surveyID = '';
+
+  constructor(private changeDetector: ChangeDetectorRef) {}
+
+  ngOnInit(): void {
+    this.fetchActiveSurveys();
+  }
+
+  private fetchActiveSurveys(): void {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        this.surveyID = survey.id;
+        this.surveyTitle = survey.questions[0].question;
+        this.changeDetector.detectChanges();
+      }
+    });
+  }
+
+  handleDismiss(): void {
+    // existing code
+  }
+
+  handleSubmit(value: number): void {
+    // existing code
+  }
+}
+```
+
+#### 3. Add the logic for displaying and hiding it.
+
+We want to make sure we don't show the survey again to users who have either submitted or dismissed it. We use [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store this data and use it to check whether to show the survey or not.
+
+```typescript file=app.component.ts
+// rest of your imports and code
+
+export class AppComponent {
+  title = 'angular-surveys';
+  showSurvey = false; // updated to false
+
+  // rest of code
+
+  private fetchActiveSurveys(): void {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        this.surveyID = survey.id;
+        this.surveyTitle = survey.questions[0].question;
+        this.checkSurveyInteraction();
+        this.changeDetector.detectChanges();
+      }
+    });
+  }
+
+  private checkSurveyInteraction(): void {
+    const hasInteractedWithSurvey = localStorage.getItem(`hasInteractedWithSurvey_${this.surveyID}`);
+    this.showSurvey = !hasInteractedWithSurvey;
+    this.changeDetector.detectChanges();
+  }
+
+  handleDismiss(): void {
+    this.showSurvey = false;
+    localStorage.setItem(`hasInteractedWithSurvey_${this.surveyID}`, 'true');
```

Comment:
Should we use the same localStorage format that PostHog does?

It looks like `seenSurvey_${this.surveyID}` 

Don't feel strongly.

Reply:
Dont think it really matters tbh. This string can be anything, as long as it's unique